packages/mui-base/src/utils/useScrollLock.ts

```diff
@@ -10,63 +9,45 @@ const activeLocks = new Set<string>();
  * @param enabled - Whether to enable the scroll lock.
  */
 export function useScrollLock(enabled: boolean = true) {
-  // Based on Floating UI's FloatingOverlay
-
   const lockId = useId();
+
   useEnhancedEffect(() => {
     if (!enabled) {
       return undefined;
     }
 
     activeLocks.add(lockId!);
 
-    const rootStyle = document.documentElement.style;
-    // RTL <body> scrollbar
-    const scrollbarX =
-      Math.round(document.documentElement.getBoundingClientRect().left) +
-      document.documentElement.scrollLeft;
-    const paddingProp = scrollbarX ? 'paddingLeft' : 'paddingRight';
-    const scrollbarWidth = window.innerWidth - document.documentElement.clientWidth;
+    const html = document.documentElement;
+    const rootStyle = html.style;
     const scrollX = rootStyle.left ? parseFloat(rootStyle.left) : window.scrollX;
     const scrollY = rootStyle.top ? parseFloat(rootStyle.top) : window.scrollY;
-
-    rootStyle.overflow = 'hidden';
-
-    if (scrollbarWidth) {
-      rootStyle[paddingProp] = `${scrollbarWidth}px`;
-    }
-
-    // Only iOS doesn't respect `overflow: hidden` on document.body, and this
-    // technique has fewer side effects.
-    if (isIOS()) {
-      // iOS 12 does not support `visualViewport`.
-      const offsetLeft = window.visualViewport?.offsetLeft || 0;
-      const offsetTop = window.visualViewport?.offsetTop || 0;
-
-      Object.assign(rootStyle, {
-        position: 'fixed',
-        top: `${-(scrollY - Math.floor(offsetTop))}px`,
-        left: `${-(scrollX - Math.floor(offsetLeft))}px`,
-        right: '0',
-      });
-    }
+    const offsetLeft = window.visualViewport?.offsetLeft || 0;
+    const offsetTop = window.visualViewport?.offsetTop || 0;
+
+    Object.assign(rootStyle, {
+      position: 'fixed',
```

Comment:
Having `scrollbar-gutter: stable` on `html` and no scrollable content is bugged in Chrome with this approach because of Chrome's bug:
https://issues.chromium.org/issues/40835364

This seems to work regardless of `scrollbar-gutter: stable`:

```suggestion
    position: html.scrollHeight > html.clientHeight ? 'fixed' : '',
```